_posts/2025-05-20-highlights-otp-28.md

```diff
@@ -0,0 +1,609 @@
+    ok;
+spawn_all(N) ->
+    spawn(?MODULE, worker, []),
+    spawn_all(N-1).
+````
+
+Here's the test code for the second scenario. Processes stay idle but they
+do not hibernate:
+
+````
+-module(benchmark_receive).
+-export([worker2/0, spawn_all/1]).
+
+worker2() ->
+    receive
+        _  -> ok
+    end.
+
+spawn_all(0) ->
+    timer:sleep(1000),
+    io:format("Memory usage: ~p~n", [erlang:memory()]),
+    timer:sleep(1000),
+    io:format("Memory usage after 1s: ~p~n", [erlang:memory()]),
+    ok;
+spawn_all(N) ->
+    spawn(?MODULE, worker, []),
+    spawn_all(N-1).
+````
+
+Memory usage is measured by `erlang:memory()` after 1 million processes have
+been spawned. For the final result, we take the average of two measurements.
+
+We spawn 1, 10 thousand, 100 thousand, and 1 million processes for both
+scenarios. Results are summarized in the following table:
+
+|       | Number of Processes | Memory Used (Mb)
+| ----------- | ----------- | ----------
+| Hibernated      | 1       | 44.8
+| Without `hibernate/0` | 1        | 47.0
+|
+| Hibernated      | 10,000       | 55.5
+| Without `hibernate/0`   | 10,000        | 73.4
+|
+| Hibernated      | 100,000       | 130.3
+| Without `hibernate/0`   | 100,000        | 307.1
+|
+| Hibernated      | 1,000,000       | 827.9
+| Without `hibernate/0`   | 1,000,000        | 2687.1
+
+When there is only 1 process, the memory usage reduction is not obvious
+yet. When there are 1 million mostly idle processes, that's more than 75%
+reduction in memory usage if you use `hibernate/0`!
+
+
+# Based Floating Point Literals
+
+Erlang/OTP 28 extends its floating point syntax to support floating point
+literals using any bases, similar to Ada and C99/C++17. This is based on
+[EEP-75](https://www.erlang.org/eeps/eep-0075).
+
+In Erlang, you can already write integers in different bases:
+
+````
+1> 2#100.
+4
+2> 3#100.
+9
+````
+
+Now, you can do the same with floating point numbers:
+
+````
+1> 2#0.011.
+0.375
+2> 3#0.011.
+0.14814814814814814
+3> 16#0.011#e5.
+4352.0
+````
+
+Such an exact representation of floating point numbers is especially useful
+in code generating tools. With only the base 10 representation, it is difficult
+to convert floats from and to other bases without precision loss. With
+based literals, you can even preserve bit level precision. For example,
+`2#0.10101#e8` represents the exact layout of a binary float.
+
+# Warnings for Use of Old-Style Catch
+
+Erlang/OTP 28 introduces a warning for using the old style `catch Expr`,
+instead of `try ... catch ... end`.
+
+The more simplistic `catch Expr` has some subtle behaviors, especially
+in corner cases. Those unwanted behaviors can make debugging and maintaining
+code containing them more difficult. Using its alternative `try ... catch ... end`
+offers better clarity.
+
+To facilitate the eventual removal of the old-style catch, the compiler
+now has an option `warn_deprecated_catch`. It can be enabled on the project
```

Comment:
```suggestion
In a future release, the use of the old `catch` construct will by default result in compiler warnings, and
possibly removed altogether in the far future.

To facilitate removing usages of the old-style `catch`, the compiler
now has an option `warn_deprecated_catch`. It can be enabled on the project
```